internal/repository/memory_test.go

```diff
@@ -353,12 +354,12 @@ func TestInMemoryRepositoryListing(t *testing.T) {
 	assert.True(t, canList)
 
 	foo, _ := storage.ParseURI("mem:///foo")
-	canList, err = storage.CanList(foo)
```

Comment:
foo -> empty seems wrong

Reply:
Can you please take that over and change if needed? Thanks.

Reply:
I don't understand this question - do you mean you do not want to make the required change?

Reply:
> I don't understand this question

And I don't really understand what you would consider a typo and would you would like to see changed.

My impression is that you would like to change `foo` to `empty` in line 356. Is that correct? 

> - do you mean you do not want to make the required change?

No not at all. This was meant as a friendly offer to you, kindly asking you for your help here. I'm not sure what's correct here, so can you please take over (or alternatively, tell me what to do here).

Hope this helps for clarify the issue.

Reply:
OK, in more detail:

You have taken the line:
`canList, err = storage.CanList(foo)`
and replaced it with:
`canList, err = storage.CanList(empty)`.

So now the test is not actually validating that `foo` can be listed.

Reply:
> OK, in more detail:
> 
> You have taken the line: `canList, err = storage.CanList(foo)` and replaced it with: `canList, err = storage.CanList(empty)`.
> 
> So now the test is not actually validating that `foo` can be listed.

Mea culpa, I was on the wrong branch! After switching to the correct branch, things are evident to me now 😄.
I fixed the typo by adding an additional commit. This should resolve the issue.